packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -393,6 +394,17 @@ abstract class FlutterCommand extends Command<void> {
     );
   }
 
+  void addDartObfuscationOption() {
+    argParser.addFlag(FlutterOptions.kDartObfuscationOption,
+      help: 'In a release build, this flag removes identifiers and replaces them '
+        'with randomized values for the purposes of source code obfuscation. The '
+        'mapping between the values and the original identifiers is stored the '
+        'symbol map created with the "--split-debug-info" option. For an app built '
```

Comment:
What happens when someone passes --obfuscate but not --split-debug-info?

Reply:
We don't output the obfuscation map. I think this is okay for now, since we don't actually have a tool to de-obfuscate a stack trace, only `flutter symbols`. I can document that this should always be combined with split-debug-info

Reply:
If it should always be paired with `--split-debug-info` then it would be good to give an error otherwise, or fill in a default value for it, I think.

Reply:
Updated so that obfuscate requires split-debug-info